### PR TITLE
Fix bug #200: Update the logic for sync

### PR DESCRIPTION
**Pull request by zjsunzone**
Notes:
1.Log level adjustment prevents excessive printing.
2.Update the request parameters of the block download in the synchronization logic.
3.Remove invalid code.

### CODE REVIEW

**Comment by zjsunzone** [comment]
> Notes:
> 1.Log level adjustment prevents excessive printing.
> 2.Update the request parameters of the block download in the synchronization logic.
> 3.Remove invalid code.

File: consensus/cbft/sync_process.go [hunk 1]
```diff
@@ -521,6 +463,26 @@ func (cbft *Cbft) OnPong(nodeID string, netLatency int64) error {
...
+// BlockExists is used to query whether the specified block exists in this node.
+func (cbft *Cbft) BlockExists(blockNumber uint64, blockHash common.Hash) error {
+	if (blockHash == common.Hash{}) {
+		return fmt.Errorf("invalid blockHash")
+	}
+	block := cbft.blockTree.FindBlockByHash(blockHash)
+	if block == nil {
+		block = cbft.blockChain.GetBlock(blockHash, blockNumber)
+	}
+	if block == nil {
```
Comment (guiting-platon) [inline consensus/cbft/sync_process.go]: 两次nil判断是否可以合在一起？

File: consensus/cbft/validator/validator.go [hunk 2]
```diff
@@ -409,14 +409,7 @@ func (vp *ValidatorPool) validatorList(blockNumber uint64) []discover.NodeID {
...
 func (vp *ValidatorPool) VerifyHeader(header *types.Header) error {
-	recPubKey, err := crypto.Ecrecover(header.SealHash().Bytes(), header.Signature())
-	if err != nil {
-		return err
-	}
-	var nodeID discover.NodeID
-	copy(nodeID[:], recPubKey[1:])
-
-	_, err = vp.getValidatorByNodeID(header.Number.Uint64(), nodeID)
+	_, err := crypto.Ecrecover(header.SealHash().Bytes(), header.Signature())
```
Comment (guiting-platon) [inline consensus/cbft/validator/validator.go]: header会存在空的可能吗？

File: core/blockchain_cache.go [hunk 3]
```diff
@@ -119,9 +119,13 @@ func (pbc *BlockChainCache) WriteReceipts(sealHash common.Hash, receipts []*type
 	pbc.receiptsMu.Lock()
 	defer pbc.receiptsMu.Unlock()
 	obj, exist := pbc.receiptsCache[sealHash]
-	if exist && obj.blockNum == blockNum {
-		obj.receipts = append(obj.receipts, receipts...)
-	} else if !exist {
+	if exist {
+		// FIXME: removing in productive environment
+		// Only for test
+		if types.DeriveSha(types.Receipts(obj.receipts)) != types.DeriveSha(types.Receipts(receipts)) {
```
Comment (guiting-platon) [inline core/blockchain_cache.go]: 生产环境会删除这些代码？如果不需要，不要提交

File: consensus/cbft/validator/validator.go [hunk 4]
```diff
@@ -409,14 +409,7 @@ func (vp *ValidatorPool) validatorList(blockNumber uint64) []discover.NodeID {
...
 	if err != nil {
 		return err
```
Comment (guiting-platon) [inline consensus/cbft/validator/validator.go]: 最好输出错误信息cbft.log.Error(“”）

File: core/blockchain_cache.go (hunk 3 above)
Comment (TraceBundy) [reply core/blockchain_cache.go]: 现在测试阶段，不是主分支，防止后面开发改动出现问题

File: consensus/cbft/validator/validator.go (hunk 4 above)
Comment (TraceBundy) [reply consensus/cbft/validator/validator.go]: 内部模块不需要打印错误吧，返回上层打印合适吧

File: consensus/cbft/sync_process.go (hunk 1 above)
Comment (zjsunzone) [reply consensus/cbft/sync_process.go]: This is two logical branches and does not need to be merged.

Comment (zjsunzone) [reply consensus/cbft/sync_process.go]: Optimized.

File: consensus/cbft/validator/validator.go (hunk 4 above)
Comment (zjsunzone) [reply consensus/cbft/validator/validator.go]: The error will be output by the top of the call chain.

**Pull request merged**